runtime/src/main/cpp/Memory.cpp

```diff
@@ -2458,6 +2464,17 @@ bool clearSubgraphReferences(ObjHeader* root, bool checked) {
   return true;
 }
 
+inline void freezeContainer(ContainerHeader* container) {
+  // Run class-specific pre-freezing code.
+  traverseContainerObjects(container, [](ObjHeader* obj) {
+    if (obj->type_info() == theWorkerBoundReferenceTypeInfo) {
+      FreezeWorkerBoundReference(obj);
```

Comment:
I'm afraid doing this here might be incorrect.
For `freezeAcyclic` it might result into some objects not being unmarked.
For `freezeCyclic` -- incorrectly freezing WBR's value etc.

Generally the entire freezing machinery doesn't expect object graph to change.